### ISSUE TITLE
`ErrorRegion`: Accessibility issue with `label={undefined}`

### ISSUE THREAD

**Issue by grigasp** [iTwin/stratakit#961]
The docs for `ErrorRegion.Root` component say that it's valid to specify `label={undefined}` for cases when we don't want to show the errors. However, we get accessibility errors when doing so.

Repro:
1. Open https://codesandbox.io/p/sandbox/pjyvr8
2. Open the preview in new tab
3. Open Chrome DevTools -> Lighthouse
4. Generate a report

Result:
<img width="703" height="304" alt="Image" src="https://github.com/user-attachments/assets/064007e9-ebe7-40af-b175-5519cfe5e2c4" />

The same error is given by [axe](https://www.deque.com/axe/):
```json
{
  "description": "Ensure ARIA attributes are not prohibited for an element's role",
  "help": "Elements must only use permitted ARIA attributes",
  "helpUrl": "https://dequeuniversity.com/rules/axe/4.10/aria-prohibited-attr?application=playwright",
  "id": "aria-prohibited-attr",
  "impact": "serious",
  "nodes": [
    {
      "all": [],
      "any": [],
      "failureSummary": "Fix all of the following: aria-labelledby attribute cannot be used on a section with no valid role attribute.",
      "html": "<section aria-labelledby=\":r0:\" class=\"🥝ErrorRegion\" data-_sk-visible=\"false\" data-_sk-expanded=\"false\" style=\"width: 100%;\">",
      "impact": "serious",
      "none": [
        {
          "data": {
            "messageKey": "noRoleSingular",
            "nodeName": "section",
            "prohibited": [
              "aria-labelledby"
            ],
            "role": null
          },
          "id": "aria-prohibited-attr",
          "impact": "serious",
          "message": "aria-labelledby attribute cannot be used on a section with no valid role attribute.",
          "relatedNodes": []
        }
      ],
      "target": [
        "section"
      ]
    }
  ],
  "tags": [
    "cat.aria",
    "wcag2a",
    "wcag412",
    "EN-301-549",
    "EN-9.4.1.2"
  ]
}
```
